Core/Services/OfferService.cs

```diff
@@ -144,5 +144,18 @@ public async Task<PaginatedList<OfferPreviewDTO>> GetUsersOffersAsync(string use
             return PaginatedList<OfferCreateTripDTO>.Evaluate(
                 _mapper.Map<List<OfferCreateTripDTO>>(offerList), paginationFilter.PageNumber, offerListCount, totalPages);
         }
+
+        public async Task DeleteAsync(OfferIdDTO offerIdDTO, string userId)
+        {
+            var offer = await _offerRepository
+                .GetBySpecAsync(new OfferSpecification
+                .GetOpenByIdAndUserIdWithoutTrip(offerIdDTO.OfferId, userId));
+
+            ExceptionMethods.OfferNullCheck(offer);
+            ExceptionMethods.PointNullCheck(offer.Point);
```

Comment:
I don't think we should check if offer.Point is not null because "OfferPointId" field should not be null and if we delete any point from DB it will automatically delete the related offer if it is